Mathlib/Analysis/SpecialFunctions/Log/ENNRealLogExp.lean

```diff
@@ -0,0 +1,170 @@
+  · simp [hx_zero]
+  have hx_pos : 0 < x.toReal := ENNReal.toReal_pos hx_zero hx_top
+  rw [← ENNReal.ofReal_toReal hx_top, log_ofReal_of_pos hx_pos, exp_coe, Real.exp_log hx_pos]
+
+end LogExp
+
+section Log
+
+@[simp]
+lemma log_lt_log_iff {x y : ℝ≥0∞} : log x < log y ↔ x < y := by
+  refine ⟨fun h ↦ ?_, fun h ↦ log_strictMono h⟩
+  rw [← exp_log x, ← exp_log y]
+  exact exp_strictMono h
+
+@[simp] lemma bot_lt_log_iff {x : ℝ≥0∞} : ⊥ < log x ↔ 0 < x := log_zero ▸ @log_lt_log_iff 0 x
+
+@[simp] lemma log_lt_top_iff {x : ℝ≥0∞} : log x < ⊤ ↔ x < ⊤ := log_top ▸ @log_lt_log_iff x ⊤
+
+@[simp] lemma log_lt_zero_iff {x : ℝ≥0∞} : log x < 0 ↔ x < 1 := log_one ▸ @log_lt_log_iff x 1
+
+@[simp] lemma zero_lt_log_iff {x : ℝ≥0∞} : 0 < log x ↔ 1 < x := log_one ▸ @log_lt_log_iff 1 x
+
+@[simp]
+lemma log_le_log_iff {x y : ℝ≥0∞} : log x ≤ log y ↔ x ≤ y := by
+  refine ⟨fun h ↦ ?_, fun h ↦ log_monotone h⟩
+  rw [← exp_log x, ← exp_log y]
+  exact exp_monotone h
+
+@[simp] lemma log_le_zero_iff {x : ℝ≥0∞} : log x ≤ 0 ↔ x ≤ 1 := log_one ▸ @log_le_log_iff x 1
+
+@[simp] lemma zero_le_log_iff {x : ℝ≥0∞} : 0 ≤ log x ↔ 1 ≤ x := log_one ▸ @log_le_log_iff 1 x
+
+end Log
+
+section Exp
+
+@[simp]
+lemma exp_lt_exp_iff {a b : EReal} : exp a < exp b ↔ a < b := by
+  conv_rhs => rw [← log_exp a, ← log_exp b, log_lt_log_iff]
+
+@[simp] lemma zero_lt_exp_iff {a : EReal} : 0 < exp a ↔ ⊥ < a := exp_bot ▸ @exp_lt_exp_iff ⊥ a
+
+@[simp] lemma exp_lt_top_iff {a : EReal} : exp a < ⊤ ↔ a < ⊤ := exp_top ▸ @exp_lt_exp_iff a ⊤
+
+@[simp] lemma exp_lt_one_iff {a : EReal} : exp a < 1 ↔ a < 0 := exp_zero ▸ @exp_lt_exp_iff a 0
+
+@[simp] lemma one_lt_exp_iff {a : EReal} : 1 < exp a ↔ 0 < a := exp_zero ▸ @exp_lt_exp_iff 0 a
+
+@[simp] lemma exp_le_exp_iff {a b : EReal} : exp a ≤ exp b ↔ a ≤ b := by
+  conv_rhs => rw [← log_exp a, ← log_exp b, log_le_log_iff]
+
+@[simp] lemma exp_le_one_iff {a : EReal} : exp a ≤ 1 ↔ a ≤ 0 := exp_zero ▸ @exp_le_exp_iff a 0
+
+@[simp] lemma one_le_exp_iff {a : EReal} : 1 ≤ exp a ↔ 0 ≤ a := exp_zero ▸ @exp_le_exp_iff 0 a
+
+lemma exp_nmul (x : EReal) (n : ℕ) : exp (n * x) = (exp x) ^ n := by
+  simp_rw [← log_eq_iff, log_pow, log_exp]
+  norm_cast
+
+lemma exp_mul (x : EReal) (y : ℝ) : exp (x * y) = (exp x) ^ y := by
+  rw [← log_eq_iff, log_rpow, log_exp, log_exp, mul_comm]
+
+end Exp
+
+section OrderIso
+
+/-- `ENNReal.log` and its inverse `ENNReal.exp` are an order isomorphism between `ℝ≥0∞` and
+`EReal`. -/
+noncomputable
+def logOrderIso : ℝ≥0∞ ≃o EReal where
+  toFun := log
+  invFun := exp
+  left_inv x := exp_log x
+  right_inv x := log_exp x
+  map_rel_iff' := by simp only [Equiv.coe_fn_mk, log_le_log_iff, forall_const]
+
+@[simp] lemma logOrderIso_apply (x : ℝ≥0∞) : logOrderIso x = log x := rfl
+@[simp] lemma logOrderIso_symm_apply (x : EReal) : logOrderIso.symm x = exp x := rfl
+
+end OrderIso
+
+section Continuity
+
+/-- `log` as a homeomorphism. -/
+noncomputable def log_homeomorph : ℝ≥0∞ ≃ₜ EReal := logOrderIso.toHomeomorph
```

Comment:
```suggestion
noncomputable def logHomeomorph : ℝ≥0∞ ≃ₜ EReal := logOrderIso.toHomeomorph
```
This is probably a better name, as this is a single unit, so no underscore in it. No need to deprecate the previous version as it was introduced very recently. Same thing for the exponential.